mantle/kola/tests/upgrade/basic.go

```diff
@@ -305,32 +306,25 @@ func runFnAndWaitForRebootIntoVersion(c cluster.TestCluster, m platform.Machine,
 	}
 }
 
-// getZincatiMetrics retrieves service metrics from Zincati over the local
-// Unix socket.
-func getZincatiMetrics(c cluster.TestCluster, m platform.Machine) string {
-	// do this in a loop in case it was just started and hasn't created the
-	// socket yet.
-	for i := 0; i < 10; i++ {
-		if _, err := c.SSHf(m, "test -S %s", zincatiMetricsSocket); err == nil {
-			break
-		}
-		time.Sleep(time.Second)
-	}
-	// either it exists, or we fail trying now
-	return string(c.MustSSHf(m, "sudo socat - UNIX-CONNECT:%s", zincatiMetricsSocket))
-}
-
 // ensureZincatiUpdatesEnabled checks Zincati metrics to make sure that
 // auto-updates logic is enabled.
 func ensureZincatiUpdatesEnabled(c cluster.TestCluster, m platform.Machine) error {
-	for i := 0; i < 10; i++ {
-		metrics := getZincatiMetrics(c, m)
-		if strings.Contains(metrics, "zincati_update_agent_updates_enabled 1") {
+	metricsCheck := func() error {
+		metrics, err := c.SSHf(m, "sudo socat - UNIX-CONNECT:%s", zincatiMetricsSocket)
+		if err != nil {
+			return err
+		}
+		if strings.Contains(string(metrics), "zincati_update_agent_updates_enabled 1") {
 			return nil
 		}
-		time.Sleep(time.Second)
+		return errors.New("failed to find 'zincati_update_agent_updates_enabled 1' on Zincati metrics.promsock")
+	}
+
+	if err := mantleUtil.Retry(10, 6*time.Second, metricsCheck); err != nil {
```

Comment:
Actually, this looks like it in fact changed the effective timeout from 100s to 60s, no? Before we had a 10-try loop within a 10-try loop, with 1s delays, and now we have a single 10 loop with 6s delay.

So maybe here, let's do e.g.

```suggestion
	if err := mantleUtil.Retry(20, 6*time.Second, metricsCheck); err != nil {
```

?